packages/merge/src/modules/merge/sagas.js

```diff
@@ -28,14 +28,54 @@ export function* loadSourceData(selection) {
   const response = yield call(rest.requestSaga, resource, options)
   const {body: sourceData} = response
   yield put(actions.setSourceData(sourceData))
+  yield put(actions.setTargetEntity(sourceData.entities[0].key))
 }
 
 export function* executeMerge() {
-  const body = {} // TODO: Implement
+  const {selection, selected, sourceData} = yield select(mergeSelector)
+
+  const paths = {}
+  for (const [name, entityKey] of Object.entries(selected.single)) {
```

Comment:
i would move this into a separate function. The code would be better readable and better testable.